downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/PathwaySummationMappingFile.java

```diff
@@ -0,0 +1,68 @@
+package org.reactome.release.downloadDirectory;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.GKInstance;
+import static org.gk.model.ReactomeJavaConstants.*;
+import org.gk.persistence.MySQLAdaptor;
+
+public class PathwaySummationMappingFile {
+	private static final Logger logger = LogManager.getLogger();
+	private static final String pathwaySummationFilename = "pathway2summation.txt";
+	
+	@SuppressWarnings("unchecked")
+	public static void execute(MySQLAdaptor dba, String releaseNumber) throws Exception {
+		
+		logger.info("Running PathwaySummationMappingFile step");
+		// Get all Pathway instances
+		Collection<GKInstance> pathwayInstances = dba.fetchInstancesByClass(Pathway);
+		Set<String> rowHash = new HashSet<String>();
+		//Create file
+		logger.info("Generating pathway2summation.txt file...");
+		File pathwaySummationFile = new File(pathwaySummationFilename);
+		pathwaySummationFile.delete();
+		pathwaySummationFile.createNewFile();
+		for (GKInstance pathwayInst : pathwayInstances) 
+		{
+			boolean containsHumanEntry = false;
+			// Check that at least one of the Species instances is for Homo sapiens
+			for (GKInstance speciesInst: (Collection<GKInstance>) pathwayInst.getAttributeValuesList(species)) 
+			{
+				if (speciesInst.getDisplayName().equals("Homo sapiens")) 
+				{
+					containsHumanEntry = true;
+				}
+			}
+			if (containsHumanEntry) 
+			{
+				// Build line of file
+				String nameString = pathwayInst.getAttributeValue(name).toString();
+				String stableId = ((GKInstance) pathwayInst.getAttributeValue(stableIdentifier)).getAttributeValue(identifier).toString();
+				for (GKInstance summationInst : (Collection<GKInstance>) pathwayInst.getAttributeValuesList(summation)) 
+				{
+					String textString = summationInst.getAttributeValue(text).toString().replaceAll(" +", " ").replaceAll("\n", "");
```

Comment:
The loop might be simpler and more readable if it iterates over the summation text directly rather than the summation instance:
```
for (String textString : getSummationsText(pathwayInst)) {
...
}


@SuppressWarnings("unchecked")
private static List<String> getSummationsText(GKInstance pathway) {
    Collection<GKInstance> summationInstances;
    try {
        summationInstances = (Collection<GKInstance>) pathway.getAttributeValuesList(summation);
    } catch (Exception e) {
        logger.warn("Could not get summation instances for pathway " + pathway.getExtendedDisplayName(), e);
        return new ArrayList<>();
    }

    return summationInstances
        .stream()
        .map(summationInstance -> getSummationText(summationInstance))
        .filter(text -> !text.isEmpty())
        .collect(Collectors.toList());
}

private static String getSummationText(GKInstance summationInstance) {
    try {
        return summationInstance.getAttributeValue(text).toString()
            .replaceAll(" +", " ")
            .replaceAll("\n", "");
    } catch (Exception e) {
        logger.warn("Could not get summation text for summation instance "
            + summationInstance.getExtendedDisplayName(), e);
        return "";
    }
}
```

Reply:
I feel like this adds many lines, no? Is it an explicitly better way of doing it since its streaming?

Reply:
I don't think it's always better. I think in this case it's more a matter of style/personal preference.

Reply:
On second thought, this doesn't seem necessary.